src/Minisite.php

```diff
@@ -1,31 +1,214 @@
+    MinisiteValidator::validateFiles($files, $contentExtensions);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getIndexAssetUri() {
+    foreach ($this->assets as $asset) {
+      if ($asset->isIndex()) {
+        return $asset->getSource();
+      }
+    }
+
+    throw new InvalidContentArchiveException(sprintf('Missing index file %s', MinisiteAsset::INDEX_FILE));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getCommonArchiveDir() {
+    return \Drupal::config('system.file')->get('default_scheme') . '://' . MinisiteInterface::ARCHIVE_UPLOAD_DIR;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getCommonAssetDir() {
+    return \Drupal::config('system.file')->get('default_scheme') . '://' . MinisiteInterface::ASSET_DIR;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function supportedArchiveExtensions() {
+    return explode(' ', MinisiteInterface::SUPPORTED_ARCHIVE_EXTENSIONS);
+  }
+
+  /**
+   * Process archive.
+   *
+   * @throws \Drupal\minisite\Exception\UnableToExtractArchiveException
+   */
+  protected function processArchive() {
+    $archiver = self::getArchiver($this->archiveFile->getFileUri());
+
+    $asset_directory = $this->prepareAssetDirectory();
+    $files = $archiver->listContents();
+    $archiver->extract($asset_directory);
+
+    foreach ($files as $file) {
+      $asset = new MinisiteAsset($asset_directory . DIRECTORY_SEPARATOR . $file);
+      $this->assets[] = $asset;
+    }
+  }
+
+  /**
+   * Prepare asset directory.
+   *
+   * @return string
+   *   Prepared asset directory.
+   *
+   * @throws \Drupal\minisite\Exception\AssetException
+   *   When unable to prepare asset.
+   */
+  protected function prepareAssetDirectory() {
+    $suffix = $this->archiveFile->get('uuid')->value;
+    $dir = self::getCommonAssetDir() . DIRECTORY_SEPARATOR . $suffix;
+    if (!\Drupal::service('file_system')->prepareDirectory($dir, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS)) {
+      throw new AssetException(sprintf('Unable to prepare asset directory "%s"', $dir));
+    }
+
+    return $dir;
+  }
+
+  /**
+   * Instantiate an archiver instance base on the file uir and name.
+   *
+   * @param string $uri
+   *   URI of the file to instantiate the archiver manager with.
+   * @param string $filename
+   *   (optional) File name to pass to the archiver manager. if not provided,
+   *   the filename will be extracted from the $uri.
+   *
+   * @return \Drupal\Core\Archiver\ArchiverInterface
+   *   The archiver instance.
+   */
+  protected static function getArchiver($uri, $filename = NULL) {
+    $filename_real = \Drupal::service('file_system')->realpath($uri);
+    $filename = $filename ? $filename : \Drupal::service('file_system')->basename($uri);
+
+    try {
+      /** @var \Drupal\Core\Archiver\ArchiverInterface $archiver */
+      $archiver = \Drupal::getContainer()->get('plugin.manager.minisite_archiver')->getInstance([
```

Comment:
this is using our own version of ArchiverManager  - see MinisteArchiverManager for more explanation